NineChronicles.DataProvider/RenderSubscriber.cs

```diff
@@ -147,15 +223,112 @@ protected override Task ExecuteAsync(CancellationToken stoppingToken)
                         {
                             MySqlStore.DeleteCombinationEquipment(combinationEquipment.Id);
                             Log.Debug("Deleted CombinationEquipment action in block #{index}", ev.BlockIndex);
+                            string avatarName = ev.OutputStates.GetAvatarState(combinationEquipment.AvatarAddress).name;
+                            var inventory = ev.OutputStates.GetAvatarState(combinationEquipment.AvatarAddress).inventory;
+                            if (inventory is { } inventoryNotNull)
+                            {
+                                ProcessEquipments(
+                                    ev.Signer,
+                                    combinationEquipment.AvatarAddress,
+                                    inventoryNotNull,
+                                    avatarName);
+                            }
+
+                            Log.Debug(
+                                "Reverted avatar {address}'s equipments in block #{index}",
+                                combinationEquipment.AvatarAddress,
+                                ev.BlockIndex);
                         }
 
                         if (ev.Action is ItemEnhancement itemEnhancement)
                         {
                             MySqlStore.DeleteItemEnhancement(itemEnhancement.Id);
                             Log.Debug("Deleted ItemEnhancement action in block #{index}", ev.BlockIndex);
+                            string avatarName = ev.OutputStates.GetAvatarState(itemEnhancement.avatarAddress).name;
+                            var inventory = ev.OutputStates.GetAvatarState(itemEnhancement.avatarAddress).inventory;
+                            if (inventory is { } inventoryNotNull)
+                            {
+                                ProcessEquipments(
```

Comment:
IMO, it would be better if we specify itemId for this action and migrate that item only.